src/teleport.cpp

```diff
@@ -199,14 +199,17 @@ bool teleport::teleport_to_point( Creature &critter, tripoint target, bool safe,
     }
     //player and npc exclusive teleporting effects
     if( p ) {
-        g->place_player( p->pos() );
         if( add_teleglow ) {
             p->add_effect( effect_teleglow, 30_minutes );
         }
     }
     if( c_is_u ) {
         g->update_map( *p );
```

Comment:
If the target is player, I think it is better to leave `place_player` just in case
```suggestion
        g->place_player( p->pos() );
        g->update_map( *p );
```